Xlib/support/unix_connect.py

```diff
@@ -127,11 +112,37 @@ def get_socket(dname, protocol, host, dno):
         raise error.DisplayConnectionError(dname, str(val))
 
     # Make sure that the connection isn't inherited in child processes.
-    fcntl.fcntl(s.fileno(), F_SETFD, FD_CLOEXEC)
+    _ensure_not_inheritable(s)
 
     return s
 
 
+def _ensure_not_inheritable(sock):
+    # According to PEP446, in Python 3.4 and above,
+    # it is not inherited in child processes by default.
+    # However, just in case, we explicitly make it non-inheritable.
+    # Also, we don't use the code like the following,
+    # because there would be no possibility of backporting to past versions.
+    #   if sys.version_info.major == 3 and sys.version_info.minor >= 4:
+    #       sock.set_inheritable(False)
+    #       return
+    # We just check if the socket has `set_inheritable`.
+    if hasattr(sock, 'set_inheritable'):
+        sock.set_inheritable(False)
+        return
+
+    # On Windows,
+    # Python doesn't support fcntl module because Windows doesn't have fcntl API.
+    # At least by not importing fcntl, we will be able to import python-xlib on Windows.
+    if platform.system() == 'Windows':
+        # so.. unfortunately, for Python 3.3 and below, on Windows,
+        # we can't make sure that the connection isn't inherited in child processes for now.
+        return
+
+    import fcntl
+    fcntl.fcntl(sock.fileno(), fcntl.F_SETFD, fcntl.FD_CLOEXEC)
```

Comment:
Suggestion:
```suggestion
# According to PEP446, in Python 3.4 and above,
# it is not inherited in child processes by default.
# However, just in case, we explicitly make it non-inheritable.
# We just check if the socket has `set_inheritable` to having to do a specific version check.
if hasattr(sock, 'set_inheritable'): 
    def _ensure_not_inheritable(sock):
        sock.set_inheritable(False)
else:
    try:
        import fcntl

        FD_CLOEXEC = getattr(fcntl, 'FD_CLOEXEC', 1)
        F_SETFD = getattr(fcntl, 'F_SETFD', 2)
        def _ensure_not_inheritable(sock):
            fcntl.fcntl(sock.fileno(), fcntl.F_SETFD, fcntl.FD_CLOEXEC)
    except ImportError:
        def _ensure_not_inheritable(sock):
          pass
```

Reply:
This suggestion breaks the CI.